noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
     traits::{Serialize, ToField},
 };
 
-pub struct PrivateLogContentConstraintsEnum {
-    /// The contents of the log are entirely unconstrained, and could have any values.
+/// Specifies the configuration parameters for message delivery across three key dimensions:
```

Comment:
Some extra thoughts re. this text. There are some inaccuracies, but other than that it is not incorrect. I don't think however that it is very educational - even the ordering of concepts is odd. For example, I'd first mention that delivery can be done on chain or out of band, and that if done out of band no constraints can exist on the message because it is never stored anywhere anyway. Then we can talk about encryption for the on-chain case, which is the only place where it is relevant. But doing encryption and then delivery just seems odd.

Then there's the second section at the end, which just repeats the three decisions that are explained here, except onchain unconstrained is mistakenly discarded as useless. Both can just be merged.